v-next/hardhat/src/types/network.ts

```diff
@@ -40,11 +40,16 @@ export type DefaultChainType = ChainTypeConfig extends {
 interface to allow the user to change the default chain type. */
 export interface ChainTypeConfig {}
 
+export type NetworkConnectionParams<
+  ChainTypeT extends ChainType | string = DefaultChainType,
+> = NetworkConfigOverride & {
+  networkName?: string;
```

Comment:
I think we should have two overloads of connect.
`connect(network?: string)`

and this one with object. 

A few comments about the object:
- Can we do `network` instead of network name?
- move NetworkConfigOverride to an `override` field